stdlib/pathlib.pyi

```diff
@@ -198,6 +200,10 @@ class Path(PurePath):
         def write_text(self, data: str, encoding: str | None = ..., errors: str | None = ...) -> int: ...
     if sys.version_info >= (3, 8):
         def link_to(self, target: StrPath | bytes) -> None: ...
+    if sys.version_info >= (3, 12):
+        def walk(
+            self: Self, top_down: bool = ..., on_error: _OnError | None = ..., follow_symlinks: bool = ...
+        ) -> Iterator[tuple[Self, list[AnyStr], list[AnyStr]]]: ...
```

Comment:
Since `pathlib` only supports string paths (not bytes paths), I think these are guaranteed to be lists of `str`s. This is different to `os.walk`, which supports string paths and bytes paths.

```suggestion
        ) -> Iterator[tuple[Self, list[str], list[str]]]: ...
```